aws/resource_aws_workspaces_directory.go

```diff
@@ -104,6 +104,51 @@ func resourceAwsWorkspacesDirectory() *schema.Resource {
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
 			"tags": tagsSchema(),
+			"workspace_access_properties": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"device_type_android": {
```

Comment:
lets keep the API structure here and use string and validate values.

Reply:
I don't mind, but in this case `self_service_permissions` block should be changed too as far as they have the same semantic. If it makes sense to you, please could you suggest the best way to handle `self_service_permissions` backward compatibility?

Reply:
It may require a breaking change in the future but lets keep to the scope of this PR as this is a new block added. from my experience maintainers usually prefer(and i agree) to keep the original type used in the SDK. we dont know the reason why this was not bool and in the future this opens a door for more values (as unlikely as it looks now) and it means breaking this interface or adding extra arguments.

Reply:
Make total sense to me. Thank you, Ilia 🙇 Revamping...

Reply:
Done.

Reply:
Whats the default based on?
https://docs.aws.amazon.com/workspaces/latest/api/API_WorkspaceAccessProperties.html doesnt specify a value.
if this is indeed the defaults returned lets also add a check to the base test.

Reply:
Indeed API doesn't specify the defaults, here is my best effort to guess the sensible values. Is it OK to rely on the current **non-specified** default values from API response?

Reply:
Yeah, they are defacto defaults. they still may change but more unlikely. just adding the test to basic would be fine we can use optional with default. i would avoid computed to catch config drift. so the way it was before is good just keep the check  for basic test as well.

sorry for the multiple cycles if i wasnt clear.

Reply:
No worries, I appreciate your thoughtful review 🙇 
- Removed `Default` schema options in favor of API default values
- Added default values assertions to the basic test

Could you elaborate on `Computed` schema option removal? If I understand correctly, user may skip `workspace_access_properties` attribute configuration and API returns default values, which are populated to state on `Read`. This kind of behavior should be marked as `Computed`. Also, [`basic` acc test](https://github.com/hashicorp/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#basic-acceptance-tests) should contain assertions exactly for `Computed` + `Optional` attributes.

Reply:
Doesnt the current setting cause config drift?
there should be a computed: true or default value.

Reply:
`workspace_access_properties` doesn't have a default value right now. The plan is empty at the end of each test step.